src/main/java/uk/gov/hmcts/divorce/solicitor/event/page/Applicant2ServiceDetails.java

```diff
@@ -94,4 +107,30 @@ public void addTo(final PageBuilder pageBuilder) {
                     "This address will be used to notify them about the application")
             .done();
     }
+
+    public AboutToStartOrSubmitResponse<CaseData, State> midEvent(CaseDetails<CaseData, State> details,
+                                                                  CaseDetails<CaseData, State> detailsBefore) {
+        CaseData caseData = details.getData();
+        Applicant applicant2 = caseData.getApplicant2();
+        List<String> validationErrors = new ArrayList<>();
+
+        boolean validApplicantEmail;
+        if (applicant2.getEmail() != null && !applicant2.getEmail().isEmpty()) {
+            validApplicantEmail = EmailValidator.getInstance().isValid(applicant2.getEmail());
+            if (!validApplicantEmail) {
+                validationErrors.add(INVALID_APPLICANT_EMAIL_ERROR);
+            }
+        }
+
+        if (applicant2.isRepresented()) {
+            boolean validSolicitorEmail = EmailValidator.getInstance().isValid(applicant2.getSolicitor().getEmail());
```

Comment:
applicant 2 sol email is optional ?

Reply:
App2 sol email is only requested (mandatory) if app2 is represented.